pydrawise/client.py

```diff
@@ -370,3 +356,101 @@ async def get_watering_report(
+                    **{
+                        "from": int(start.timestamp()),
+                        "until": int(end.timestamp()),
+                    }
+                ).select(*get_selectors(self._schema, WateringReportEntry)),
+            ),
+        )
+        result = await self._query(selector)
+        return Hydrawise._prune_watering_report_entries(
+            deserialize(
+                list[WateringReportEntry], result["controller"]["reports"]["watering"]
+            ),
+            start,
+            end,
+        )
+
+    async def get_water_use_summary(
+        self, controller: Controller, start: datetime, end: datetime
+    ) -> ControllerWaterUseSummary:
+        """Calculate the water use for the given controller and time period.
+
+        :param controller: The controller whose water use to report.
+        :param start: Start time
+        :param end: End time."""
+        selector = self._schema.Query.controller(controllerId=controller.id).select(
+            self._schema.Controller.sensors.select(
+                *get_selectors(self._schema, Sensor),
+                self._schema.Sensor.flowSummary(
+                    start=int(start.timestamp()),
+                    end=int(end.timestamp()),
+                ).select(*get_selectors(self._schema, SensorFlowSummary)),
+            ),
+            self._schema.Controller.reports.select(
+                self._schema.Reports.watering(
+                    **{
+                        "from": int(start.timestamp()),
+                        "until": int(end.timestamp()),
+                    }
+                ).select(*get_selectors(self._schema, WateringReportEntry)),
+            ),
+        )
+        result = await self._query(selector)
+        summary = ControllerWaterUseSummary()
+
+        # watering report entries
+        entries = Hydrawise._prune_watering_report_entries(
+            deserialize(
+                list[WateringReportEntry], result["controller"]["reports"]["watering"]
+            ),
+            start,
+            end,
+        )
+
+        # total active water use
+        for entry in entries:
+            if (
+                entry.run_event is not None
+                and entry.run_event.zone is not None
+                and entry.run_event.reported_water_usage is not None
+            ):
+                active_use = entry.run_event.reported_water_usage.value
+                if summary.unit == "":
+                    summary.unit = entry.run_event.reported_water_usage.unit
+                summary.total_active_use += active_use
+                summary.active_use_by_zone.setdefault(entry.run_event.zone.id, 0)
+                summary.active_use_by_zone[entry.run_event.zone.id] += active_use
+
+        # total active and inactive water use
+        for sensor in result["controller"]["sensors"]:
+            if (
+                "flow meter" in sensor["model"]["name"].lower()
```

Comment:
There's an enum in the schema called `CustomSensorTypeEnum`. Can we use that instead? It has a field called `FLOW` which I assume is a flow meter sensor.

Reply:
Good catch. Wasn't aware of it.

Reply:
Fixed.